### DISCUSSION TITLE
Select which displays should participate in the "next/previous display" rotation

### DISCUSSION THREAD

**Discussion by jogloran** [rxhanson/Rectangle#422] [Ideas]
I have a display which I don't ever want Rectangle to move windows onto. 

I'd like to contribute a feature that enumerates the displays and allows the user to select which of them should be part of the next/previous window rotation. I'd be happy to work on this if it's a direction Rectangle might be interested in moving in.

**Comment by rxhanson** [comment]
Thanks for looking to contribute! I'm not opposed to merging a pull request with this feature, as long as it is just configured via terminal command and not in the UI at first. I would only consider having this as part of the UI if there's enough users that want it there.

**Reply by jogloran** [reply]
The fact that you have to select the displays from an enumeration of the available displays does make this a little unwieldy as a `defaults`-managed command, since you'd then have to specify the devices by some identifier. 

In any case, I'm happy to contribute this as a command-line feature to begin with.

**Reply by rxhanson** [reply]
Indeed it would make this feature a little unwieldy. I would be willing to merge a change to the UI as long as it is still behind a terminal flag, like todo mode: https://github.com/rxhanson/Rectangle/wiki/Todo-Mode. The simplest UI addition for this feature is most likely the addition of a menu bar menu item to toggle the status of the display. 

To get a unique identifier for the display without a UI addition, it is possible to just grab it from the logs (hold the option key in the menu bar menu), since each window action will log the display to the log viewer. You could identify the display by it's rect: origin & dimensions along with the name of the screen.
